docs/DevGuide/WritingDox.md

```diff
@@ -31,11 +31,75 @@ While the `file_name` portion is not necessary, it is useful for reducing the
 likelihood of reference collisions. You can add a table of contents using the
 Doxygen command <code>\\tableofcontents</code>. All sections, subsections,
 subsubsections, etc. are shown in the table of contents if they have a tag,
-if not they do not appear. 
+if not they do not appear.
 
 # C++ Documentation {#writing_dox_cpp_dox_help}
 
-Doxygen comments...
+We encourage you to add Doxygen documentation for any of the following
+you may have added in a new hpp file:
+
++ classes
+
++ structs
+
++ enums
+
++ functions
+
++ namespaces
+
+Documentation begins on the line immediately above the declaration with
+either a triple slash `///` or a `/*!`. Build your documentation by running
+`make doc` in your build directory. You can then view it by opening
+`docs/index.html` in a browser and using the file navigator in Doxygen to
+locate your file.
+
+In addition to providing a file directory of all the source files in SpECTRE,
+Doxygen also conveniently provides two additional organizations of the files,
+Topics and Namespaces. To ensure that your documentation is easily found from
+within Doxygen, we recommend that you add any new objects to Topics and any
+new namespaces to Namespaces.
+
+## Add your object to an existing Topic:
+
+Use the Doxygen keyword `\ingroup` followed by the name of the Topic (you can
+find the list of existing Topics in `docs/GroupDefs.hpp`).
+
+## Add a new Topic:
+
+Within `docs/GroupDefs.hpp`, add the name of your Topic (which follows the
+naming convention "YourNameForTopic" followed by "Group") among the rest,
+taking care to keep the list of Topics in alphabetical order.
+
+## Add a new namespace:
+
+In the hpp file in which the namespace is declared for the first time,
+add a Doxygen comment to the line directly above the namespace. Subsequent
+files which use this namespace will not need a Doxygen comment.
+
+We also encourage you to:
+
+## Put any mathematical expressions in your documentation into LaTeX form:
+
+Within a Doxygen comment, begin your expression with `\f$` and conclude your
+expression with `\f$`.
+
+## Put any links to papers in your documentation in tags:
+
+Use the form `<a href="http://paperlocation.etc/and/so/on">
+clickable text to link here! </a>`.
+
+## Include any pictures that aid in the understanding of your documentation:
+
+First, please compress your image to be under 130kB. As most images will be
+diagrammatical in nature this is certainly achievable!
+
+Second, add the directory that will contain your image to the `docs/Images/`
+directory. Make the lowest level directory the name of your hpp file (as it is
```

Comment:
why can't you just use the same dir format and include images with different names in the same file?

Reply:
I chose to have the directory structure as:
 - Domain

    - CoordinateMaps

      - Wedge3D

        - wedge3dimage1.png

        - wedge3dimage2.png

      - BulgedCube
        - bulgedcubeimage.png

As opposed to:

 - Domain

    - CoordinateMaps

        - wedge3dimage1.png

        - wedge3dimage2.png

        - bulgedcubeimage.png

But if there is a preference for one over the other I can change it as well.

Reply:
My preference would be for docs/Images to have the same directory structure as src and tests/Unit .  I don't see the point of having many directories with one or two files in them.  Also I would CamelCase the names of the files, dropping the word image as that is implied by the fact it is in the docs/Images directory.  @nilsdeppe @wthrowe what are your thoughts?

Reply:
Is there going to be much image reuse between different files?  If so that would be a good reason for a flatter structure.

Reply:
I will make these changes :+1: